setup.py

```diff
@@ -18,7 +18,9 @@ def read(fname):
     "tqdm>=4.41.0",
     "shap >= 0.38.1, < 0.39.0",# 0.40.0 causes issues in certain plots. For now it is excluded
     "numpy>=1.19.0",
-    "lightgbm>=3.3.0"
+    "lightgbm>=3.3.0",
```

Comment:
Since now we don't import the lightgbm at the top of the file, the 3 packages can go to the extra dependencies. In case someone wants to use one of these models in shap RFECV the users will take care of installing them since they will need to pass them.
I would put these 3 into extra dependencies. What do you think?

Reply:
Not unorthodox at all actually :) You can take this a step further and replace the import with a 'mock' instance where `.__call__` will invoke a helpful 'not installed' error only when the import is actually used. `probatus` already supports this, you can use `utils.exceptions.NotInstalledError`. See https://github.com/ing-bank/probatus/blob/5c5b0aac216881782e005e79ab47668d7a93164a/probatus/utils/exceptions.py#L57

Reply:
Hi @timvink, thanks for joining this review :). 

Sorry, I did not understand what do you mean when you say

> ... replace the import with a 'mock' instance ... 

The intention of my code is:

I am a user of Probatus that likes to use LightGBM.
I already have a virtual environment where Probatus and LightGBM are installed.
I use the class RFECV of Probatus passing the LightGBM classifier instance I have created earlier in my code.
Probatus tries to import cat boosts, but it fails because the user has installed LightGBM - this step is ignored.
Probatus tries to import LightGBM and it succeeds.
Probatus Cretes fit params compatible with LightGBM.

Now Imagine the user wants to pass something else to RFECV, like a RandomForestClassifier from sklearn. All of the imports will fail and the user will receive a model not supported error.

It seems like a legit behaviour to me.

What would you suggest instead :) ?

Reply:
No problem, still keeping an eye on the probatus development :)

I didn't look in detail enough, I thought you had a top level `try: import .. except: pass` and then `NotInstalledError` can help with a 'mock' to create a more helpfull error message.

I looked at your implementation of ` def _get_fit_params` and I think it's quite elegant actually!

Reply:
Thanks :), happy to hear that you are still hanging around ;)